infra/util/src/main/java/org/apache/shardingsphere/infra/util/spi/ShardingSphereServiceLoader.java

```diff
@@ -70,7 +80,17 @@ private Collection<T> load() {
     @SuppressWarnings("unchecked")
     public static <T> Collection<T> getServiceInstances(final Class<T> serviceInterface) {
         ShardingSphereServiceLoader<?> result = LOADERS.get(serviceInterface);
-        return (Collection<T>) (null != result ? result.getServiceInstances() : LOADERS.computeIfAbsent(serviceInterface, ShardingSphereServiceLoader::new).getServiceInstances());
+        if (null != result) {
+            return (Collection<T>) result.getServiceInstances();
+        }
+        synchronized (LOAD_LOCKS[serviceInterface.hashCode() % LOAD_LOCKS_COUNT]) {
```

Comment:
How about `synchronized (serviceInterface)`?

Reply:
Sorry, I missed this message before.

`serviceInterface` might be synchronized here and in other places, I'm afraid there might be synchronization issue, so I use dedicated lock here.

Reply:
Got it.